contracts/MakerOrders.sol

```diff
@@ -0,0 +1,79 @@
+pragma solidity 0.4.18;
+
+
+import "./Withdrawable.sol";
+
+
+contract MakerOrders is Withdrawable {
+    struct FreeOrders {
+        uint32 firstOrderId;
+        uint32 numOrders; //max is 256
+        uint8 maxOrdersReached;
+        uint256 takenBitmap;
+    }
+
+    //each maker will have orders that will be reused.
+    mapping(address => FreeOrders) public makerOrdersSell;
+    mapping(address => FreeOrders) public makerOrdersBuy;
+
+    function takeOrderId(FreeOrders storage freeOrders)
+        internal
+        onlyAdmin
+        returns(uint32)
+    {
+        uint numOrders = freeOrders.numOrders;
+        uint orderBitmap = freeOrders.takenBitmap;
+        uint bitPointer = 1;
+
+        for(uint i = 0; i < numOrders; ++i) {
+            if ((orderBitmap & bitPointer) == 0) {
+                freeOrders.takenBitmap = orderBitmap | bitPointer;
+                return(uint32(uint(freeOrders.firstOrderId) + i));
+            }
+
+            bitPointer *= 2;
+        }
+
+        freeOrders.maxOrdersReached = 1;
+
+        require(false);
```

Comment:
should decide if to use maxOrdersReached and avoid reverting.
or revert.

if we revert. maxOrdersReached will not be set anyway.

Reply:
Let's discuss this.